core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsBillingProjectComponent.scala

```diff
@@ -104,13 +103,23 @@ trait RawlsBillingProjectComponent {
      * @param rawlsProjectName
      * @return
      */
-    def loadDirectProjectMembersWithEmail(rawlsProjectName: RawlsBillingProjectName): ReadAction[Seq[RawlsBillingProjectMember]] = {
-      val usersQuery = findDirectProjectUsers(rawlsProjectName)
-      val subgroupsQuery = findDirectProjectSubgroups(rawlsProjectName)
-
-      val membersQuery = usersQuery union subgroupsQuery
-
-      membersQuery.result.map(_.map{ case(email, role) => RawlsBillingProjectMember(RawlsUserEmail(email), ProjectRoles.withName(role))})
+    def loadDirectProjectMembersWithEmail(rawlsProjectName: RawlsBillingProjectName): ReadWriteAction[Seq[RawlsBillingProjectMember]] = {
+      for {
+        roleGroupRecs <- rawlsBillingProjectGroupQuery.filter(_.projectName === rawlsProjectName.value).result
+        roleGroups <- DBIO.sequence(roleGroupRecs.map { roleRec => rawlsGroupQuery.load(RawlsGroupRef(RawlsGroupName(roleRec.groupName))).map(roleRec -> _)})
+        users <- rawlsUserQuery.load(roleGroups.flatMap(_._2.get.users))
```

Comment:
I'm struggling a little with being able to read this. It might be helpful to find a more expressive way to write `_._2`. Also, is it guaranteed that `get` won't be called on `None` here? Looking at `rawlsGroupQuery.load`, I think `None` is very possible. It would probably represent an inconsistency between the Rawls DB and LDAP which ideally won't happen, but given that we don't have distributed transactions between the two I suspect it's very possible.